Makefile

```diff
@@ -52,6 +52,7 @@ LOCALBIN ?= $(shell pwd)/bin
 KUSTOMIZE_VERSION ?= v5.4.1
 CONTROLLER_TOOLS_VERSION ?= v0.15.0
 KIND_VERSION ?= v0.23.0
+KIND_K8S_VERSION ?= v1.31.1
```

Comment:
Maybe we should change default k8s version to v1.29.3 as we use now.

Reply:
Do we need to keep `ENVTEST_K8S_VERSION`, `KIND_K8S_VERSION` and the Go client versions in sync? If there's no dependency I'd prefer to use the latest minor version i.e. 1.31

Reply:
Yes, I think we should keep the three k8s versions the same.

Reply:
Changed to `1.29.3`. I reckon we could bump to the latest minor at some point soon though. I don't think there are any APIs that either Spark or the operator are using that would block us, unless there are any kubebuilder/controller-runtime dependency issues.